CONTRIBUTING.md

```diff
@@ -63,10 +63,35 @@ run-time:
 ./scripts-dev/lint.sh path/to/file1.py path/to/file2.py path/to/folder
 ```
 
-You can also provided the `-d` option, which will lint the files that have been
+You can also provide the `-d` option, which will lint the files that have been
 changed since the last git commit. This will often be significantly faster than
 linting the whole codebase.
 
+Another option is to only run types checks, with `mypy`, in a fresh virtual env:
```

Comment:
I think we can now remove this section and update the install line above to be:

```
pip install -e ".[lint,mypy]"
```

Reply:
I was on the way to do this with [c168f20](https://github.com/matrix-org/synapse/pull/8633/commits/c168f20318d3ba6dd7612cdcf9a40d88cd25e7c3) (with a worse syntax ;) ) but when I tested it, it worked without mymy. After testing again, I must have done some mistake, because after testing again, it's really needed.
So, yes, you're right.

I was wondering, about the original issue, to run isolated mypy tests, but I felt kinda "bad" while writing the instructions, that the file was gaining a lot of length and some kind of digression about only a part of the tests.
I personnaly prefer the conciseness  in that file (and I'm not attach at all to what I wrote ;) ) and that the lint script run all tests.